### PR TITLE
Fix TestBPF

### LINKED ISSUE #22078: CI: TestBPF fails with "-bpf-test-path is a required flag" in runtime DatapathPrivilegedUnitTests
[09:02:19]     bpf_test.go:56: -bpf-test-path is a required flag
	 [09:02:19] --- FAIL: TestBPF (0.00s)
	 [09:02:19] FAIL
	 [09:02:19] coverage: 6.2% of statements
	 [09:02:19] FAIL	github.com/cilium/cilium/test/bpf_tests	0.037s

Started failing consistently on `master` after https://github.com/cilium/cilium/pull/22043

Latest failure on `master`: https://jenkins.cilium.io/job/cilium-master-runtime-kernel-net-next/5238/

### PR DESCRIPTION
**Pull request by tklauser**
Fixes #22078

See commits for details.

### CODE REVIEW

**Review by aanm — APPROVED** [review]

File: .github/workflows/lint-bpf-checks.yaml [hunk 1]
```diff
@@ -133,7 +133,7 @@ jobs:
           fetch-depth: 0
       - name: Run BPF tests with code coverage reporting
         run: |
-          make -C test run_bpf_tests COVER=1 || (echo "Run 'make -C test run_bpf_tests COVER=1' locally to investigate failures"; exit 1)
```
Comment (pchaigno) [inline .github/workflows/lint-bpf-checks.yaml:136]: Do we need an issue to track this change and remember to revert once the underlying issue is fixed?

**Review by pchaigno — APPROVED** [review]

File: .github/workflows/lint-bpf-checks.yaml (hunk 1 above)
Comment (tklauser) [reply .github/workflows/lint-bpf-checks.yaml:136]: Thanks, good point. I've created https://github.com/cilium/cilium/issues/22088 and cross-referenced in a comment in the line above.

File: .github/workflows/lint-bpf-checks.yaml [hunk 2]
```diff
@@ -133,7 +133,8 @@ jobs:
           fetch-depth: 0
       - name: Run BPF tests with code coverage reporting
```
Comment (pchaigno) [inline .github/workflows/lint-bpf-checks.yaml]: This title isn't accurate anymore :sweat_smile: Not sure if it's worth fixing.

Comment (tklauser) [reply .github/workflows/lint-bpf-checks.yaml]: 🤦 I've fixed and re-pushed.

**Comment by pchaigno** [comment]
Last two pushes changed only a comment and a workflow step title. The CI job that matter here were passing before. Merging.
Thanks Tobias!

**Pull request merged**